sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -0,0 +1,792 @@
+
+// @public
+export interface PiiEntityRecognitionSuccessResult extends TextAnalysisSuccessResult {
+    readonly entities: Entity[];
+    readonly redactedText: string;
+}
+
+// @public
+export interface SentenceAssessment {
+    confidenceScores: TargetConfidenceScores;
+    isNegated: boolean;
+    length: number;
+    offset: number;
+    sentiment: TokenSentimentLabel;
+    text: string;
+}
+
+// @public
+export interface SentenceSentiment {
+    readonly confidenceScores: SentimentConfidenceScores;
+    readonly length: number;
+    readonly offset: number;
+    readonly opinions: Opinion[];
+    readonly sentiment: SentenceSentimentLabel;
+    readonly text: string;
+}
+
+// @public
+export type SentenceSentimentLabel = "positive" | "neutral" | "negative";
+
+// @public
+export type SentimentAnalysisAction = ActionPrebuilt & {
+    includeOpinionMining?: boolean;
+    stringIndexType?: StringIndexType;
+};
+
+// @public
+export type SentimentAnalysisErrorResult = TextAnalysisErrorResult;
+
+// @public
+export type SentimentAnalysisResult = SentimentAnalysisSuccessResult | SentimentAnalysisErrorResult;
+
+// @public
+export interface SentimentAnalysisSuccessResult extends TextAnalysisSuccessResult {
+    readonly confidenceScores: SentimentConfidenceScores;
+    readonly sentences: SentenceSentiment[];
+    readonly sentiment: DocumentSentimentLabel;
+}
+
+// @public
+export interface SentimentConfidenceScores {
+    // (undocumented)
+    negative: number;
+    // (undocumented)
+    neutral: number;
+    // (undocumented)
+    positive: number;
+}
+
+// @public
+export type StringIndexType = string;
+
+// @public
+export interface TargetConfidenceScores {
+    // (undocumented)
+    negative: number;
+    // (undocumented)
+    positive: number;
+}
+
+// @public
+export interface TargetSentiment {
+    readonly confidenceScores: TargetConfidenceScores;
+    readonly length: number;
+    readonly offset: number;
+    readonly sentiment: TokenSentimentLabel;
+    readonly text: string;
+}
+
+// @public
+export class TextAnalysisClient {
+    constructor(endpointUrl: string, credential: TokenCredential | KeyCredential, options?: TextAnalysisClientOptions);
+    analyze<ActionName extends "LanguageDetection">(actionName: ActionName, documents: LanguageDetectionInput[], options?: AnalyzeActionParameters<ActionName> & TextAnalysisOperationOptions): Promise<AnalyzeResult<ActionName>>;
+    analyze<ActionName extends "LanguageDetection">(actionName: ActionName, documents: string[], countryHint?: string, options?: AnalyzeActionParameters<ActionName> & TextAnalysisOperationOptions): Promise<AnalyzeResult<ActionName>>;
+    analyze<ActionName extends AnalyzeActionName = AnalyzeActionName>(actionName: ActionName, documents: TextDocumentInput[], options?: AnalyzeActionParameters<ActionName> & TextAnalysisOperationOptions): Promise<AnalyzeResult<ActionName>>;
```

Comment:
what is the difference between ActionName and AnalyzeActionName? is the later batch?

Reply:
`ActionName` is a generic type used to index into other types and by default `AnalyzeActionName` = `ActionName`.